samples/vmc/sddc/deploy_ovf_template_cloud.py

```diff
@@ -0,0 +1,140 @@
+#!/usr/bin/env python
+
+"""
+* *******************************************************
+* Copyright VMware, Inc. 2018. All Rights Reserved.
+* SPDX-License-Identifier: MIT
+* *******************************************************
+*
+* DISCLAIMER. THIS PROGRAM IS PROVIDED TO YOU "AS IS" WITHOUT
+* WARRANTIES OR CONDITIONS OF ANY KIND, WHETHER ORAL OR WRITTEN,
+* EXPRESS OR IMPLIED. THE AUTHOR SPECIFICALLY DISCLAIMS ANY IMPLIED
+* WARRANTIES OR CONDITIONS OF MERCHANTABILITY, SATISFACTORY QUALITY,
+* NON-INFRINGEMENT AND FITNESS FOR A PARTICULAR PURPOSE.
+"""
+
+__author__ = 'VMware, Inc.'
+__vcenter_version__ = 'VMware Cloud on AWS'
+
+from com.vmware.content.library_client import Item
+from com.vmware.vcenter.ovf_client import LibraryItem
+from com.vmware.vcenter_client import ResourcePool
+from vmware.vapi.vsphere.client import create_vsphere_client
+
+from samples.vsphere.common import sample_cli, sample_util
+from samples.vsphere.common.id_generator import generate_random_uuid
+from samples.vsphere.contentlibrary.lib.cls_api_helper import ClsApiHelper
+
+
+class DeployOvfTemplateCloud:
+    """
+    Demonstrates the workflow to deploy an OVF library item to
+    a resource pool in VMware Cloud on AWS.
+    Note: the sample needs an existing library item with an OVF template
+    and an existing resource pool with resources for deploying the VM.
+    """
+
+    def __init__(self):
+        parser = sample_cli.build_arg_parser()
+
+        parser.add_argument('--resource_pool_name',
+                            default='Compute-ResourcePool',
+                            help='The name of the resource pool to be used.')
+
+        parser.add_argument('--libitemname',
+                            required=True,
+                            help='The name of the library item to deploy.'
+                                 'The library item should contain an OVF package.')
+
+        args = sample_util.process_cli_args(parser.parse_args())
+
+        self.vm_id = None
+        self.vm_name = 'deployed-vm-' + str(generate_random_uuid())
+
+        self.lib_item_name = args.libitemname
+        self.resource_pool_name = args.resource_pool_name
+        self.cleardata = args.cleardata
+
+        # Connect to vAPI Endpoint on vCenter Server
+        self.client = create_vsphere_client(server=args.server,
+                                            username=args.username,
+                                            password=args.password)
+
+        self.helper = ClsApiHelper(self.client, args.skipverification)
+
+    def deploy_ovf_template(self):
+
+        # Build the deployment target with resource pool ID
+        filter_spec = ResourcePool.FilterSpec(names=set([self.resource_pool_name]))
```

Comment:
How come we don't move this to a helper function or reuse get_resource_pool function? We would need to do the same thing for other samples in VMC.

Reply:
IMHO, sample code is different from regular application code. It should be simple and has less dependencies as possible. AFAIU, users goal is to read the code and understand how API calls are made, thus it's better to demo the actual call rather than using many helper methods. That said, if some common logic is pretty complex we can definitely wrap it to a  helper method, however it probably means we need a new API to handle that logic. Does it make sense?